### PR TITLE
Remove or deprecate y<0 statements for cubic chunks

### PR DESCRIPTION
**Pull request by liach**
**The Issue**
 - #1689
 - Removed y<0 checks
 
**The Proposal**
 - Remove y<0 checks
 - Use integer min value to indicate entity gui instead
 
**Possible Side Effects**
 - Might break some safety guarantees; not sure
 
**Alternatives**
 - Some people use cubic chunks, so adding its compatibility is a nice choice. The change is not too large, either.

@Barteks2x please look at this as well.

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
Looks good.

The only thing that it may break is `setBlockState`, `getLightFor`, `setLightFor` and `getLightSubtracted` accessed **directly on `Chunk`** (all others, including `getBlockState` should be fine). If there are any places where this is done (unlikely), I would recommend using `World.isOutsideBuildHeight`.

File: src/main/java/mods/railcraft/common/gui/GuiHandler.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public static void openGui(EnumGui gui, EntityPlayer player, World world, int x,
     public static void openGui(EnumGui gui, EntityPlayer player, World world, Entity entity) {
         if (Game.isHost(world)) {
             if (gui.hasContainer()) {
-                player.openGui(Railcraft.getMod(), gui.ordinal(), world, entity.getEntityId(), -1, 0);
+                player.openGui(Railcraft.getMod(), gui.ordinal(), world, entity.getEntityId(), Integer.MIN_VALUE, 0);
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/gui/GuiHandler.java:45]: This probably works, but if it doesn't we could always pack it into to gui.ordinal() parameter. That's how Forestry sends page numbers for its GUIs.

**Review by CovertJaguar — APPROVED** [review]

File: src/main/java/mods/railcraft/common/gui/GuiHandler.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/gui/GuiHandler.java:45]: Uhh, using an `~` bitwise operation may be too confusing for coders compared to `-`, though `-` is an incorrect operation.

File: src/main/java/mods/railcraft/common/gui/GuiHandler.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public static void openGui(EnumGui gui, EntityPlayer player, World world, int x,
...
             }
         } else if (!gui.hasContainer()) {
             FMLClientHandler.instance().displayGuiScreen(player, FactoryGui.build(gui, player.inventory, entity, world, entity.getEntityId(), -1, 0));
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/gui/GuiHandler.java:48]: What about this line? I think it might only affect the TNT carts.

**Pull request merged**